src/hotspot/share/runtime/thread.hpp

```diff
@@ -739,9 +739,11 @@ class Thread: public ThreadShadow {
  public:
   volatile intptr_t _Stalled;
   volatile int _TypeTag;
-  ParkEvent * _ParkEvent;                     // for Object monitors, JVMTI raw monitors,
+  ParkEvent * volatile _ParkEvent;            // for Object monitors, JVMTI raw monitors,
                                               // and ObjectSynchronizer::read_stable_mark
-  int NativeSyncRecursion;                    // diagnostic
+
+                                              // _ParkEvent is cleared
+  bool has_terminated()                       { return Atomic::load(&_ParkEvent) == NULL; };
```

Comment:
Please change the comment to:

// Termination indicator used by the signal handler. _ParkEvent is just a convenient field
// we can NULL out after setting the JavaThread termination state (which can't itself be read 
// from the signal handler if a signal hits during the Thread destructor.

Reply:
Fixed